### PR TITLE
master lpd 7996

### PR DESCRIPTION
**Pull request by kyle-miho**
https://liferay.atlassian.net/browse/LPD-7996

cc: @drewbrokke @nicolas-sss

Nicolas can you review this when you have a chance? Thanks!
Unit tests pass locally as well

### CODE REVIEW

**Comment by kyle-miho** [comment]
ci:test:sf

**Review by drewbrokke — COMMENTED** [review]
@kyle-miho this seems good, though can you help me understand what was happening before that made the matcher necessary?

**Comment by kyle-miho** [comment]
@drewbrokke Before, the method makes an assumption than when you are performing substitution for params (for example if param#0# gets replaced into className), it assumes that the amount of substitutions always equaled the parameters.

However for parameters such as `new InfoItemReference(param#0#, param#1#) ` which is a parameter that requires multiple substitutions, which is also an edge case, there is currently only the ability to replace one of the parameters. Before I used a while loop to keep making substitutions until it no longer could, but there was a possible infinite loop with specific incorrect syntax so I decided to move to a matcher to prevent that altogether. I'm not sure if its the best possible way, but using a matcher to find the exact amount of subsitutions seemed like a good solution altogether (although I'm unsure of how much of a performance impact it makes)

**Comment by drewbrokke** [comment]
@kyle-miho thanks for the explanation. @ling-alan-huang does this solution look ok to you? If so then I'm ok to forward.

**Review by nicolas-sss — CHANGES_REQUESTED** [review]

File: modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/expected/upgrade/UpgradeCatchAllCheck.testjava [hunk 1]
```diff
@@ -535,6 +541,11 @@ public class UpgradeCatchAllCheck implements ConfigurationBeanDeclaration {
 		_commercePriceEntryLocalService.addCommercePriceEntry(externalReferenceCode, cProductId, cpInstanceUuid, commercePriceListId, price, false, promoPrice, null, serviceContext);
 	}
 
+	public void method(String className, long classPK, ThemeDisplay themeDisplay) {
```
[the remark this answers is not in the document]
Comment (nicolas-sss) [reply modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/expected/upgrade/UpgradeCatchAllCheck.testjava:544]: Good job @kyle-miho! It was an edge case. Be careful when changing the CatchAll logic, especially with its regex, as it deals with different automation patterns. But I liked the solution and as long as the test continues to pass, I don't think any other cases will have been impacted.

Also, to reinforce this on the PR validation as well, before we create a test suite case, especially for Upgrade SF checks. It will run all upgrade tests. This way we test not only the specific check but all others as well.

ci:test:hotel

**Comment by ling-alan-huang** [comment]
@kyle-miho Please see https://github.com/liferay-devtools/liferay-portal/pull/72
